server/catgenome/src/main/java/com/epam/catgenome/manager/lineage/LineageTreeManager.java

```diff
@@ -0,0 +1,266 @@
+    private static final int EDGES_FILE_COLUMNS = 4;
+    private final LineageTreeDao lineageTreeDao;
+    private final LineageTreeNodeDao lineageTreeNodeDao;
+    private final LineageTreeEdgeDao lineageTreeEdgeDao;
+    private final BiologicalDataItemManager biologicalDataItemManager;
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public LineageTree createLineageTree(final LineageTreeRegistrationRequest request) throws IOException {
+        final String nodesPath = request.getNodesPath();
+        final String edgesPath = request.getEdgesPath();
+        getFile(nodesPath);
+        getFile(edgesPath);
+        List<LineageTreeNode> nodes = readNodes(nodesPath);
+        List<LineageTreeEdge> edges = readEdges(edgesPath,
+                nodes.stream().map(LineageTreeNode::getName).collect(Collectors.toList()));
+        LineageTree lineageTree = getLineageTree(request, nodesPath, edgesPath);
+        biologicalDataItemManager.createBiologicalDataItem(lineageTree);
+        lineageTree.setBioDataItemId(lineageTree.getId());
+        lineageTree = lineageTreeDao.saveLineageTree(lineageTree);
+        for (LineageTreeNode node: nodes) {
+            node.setLineageTreeId(lineageTree.getLineageTreeId());
+        }
+        for (LineageTreeEdge edge: edges) {
+            edge.setLineageTreeId(lineageTree.getLineageTreeId());
+        }
+        nodes = lineageTreeNodeDao.save(nodes);
+        edges = lineageTreeEdgeDao.save(edges, getNodesMap(nodes));
+        lineageTree.setNodes(nodes);
+        lineageTree.setEdges(edges);
+        return lineageTree;
+    }
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public void deleteLineageTree(final long lineageTreeId) {
+        final LineageTree lineageTree = getLineageTree(lineageTreeId);
+        lineageTreeEdgeDao.deleteLineageTreeEdges(lineageTreeId);
+        lineageTreeNodeDao.deleteLineageTreeNodes(lineageTreeId);
+        lineageTreeDao.deleteLineageTree(lineageTreeId);
+        biologicalDataItemManager.deleteBiologicalDataItem(lineageTree.getBioDataItemId());
+    }
+
+    public LineageTree loadLineageTree(final long lineageTreeId) {
+        final LineageTree lineageTree = lineageTreeDao.loadLineageTree(lineageTreeId);
+        if (lineageTree == null) {
+            return null;
+        }
+        final List<LineageTreeNode> nodes = lineageTreeNodeDao.loadLineageTreeNodes(lineageTreeId);
+        final List<LineageTreeEdge> edges = lineageTreeEdgeDao.loadLineageTreeEdges(lineageTreeId);
+        lineageTree.setNodes(nodes);
+        lineageTree.setEdges(edges);
+        return lineageTree;
+    }
+
+    public List<LineageTree> loadLineageTrees(final long referenceId) {
+        final List<LineageTree> trees = new ArrayList<>();
+        final List<LineageTreeNode> referenceNodes = lineageTreeNodeDao.loadLineageTreeNodesByReference(referenceId);
+        for (LineageTreeNode referenceNode: referenceNodes) {
```

Comment:
I'd suggest to get unique tree ids at first. In most cases we won't have only one node associated with a specific reference in a tree, but still it is better to avoid tree duplication.